modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClusterCachesInfo.java

```diff
@@ -127,6 +127,8 @@
     /** Dynamic caches. */
     private final ConcurrentMap<String, DynamicCacheDescriptor> registeredCaches = new ConcurrentHashMap<>();
 
+    private final ConcurrentMap<Integer, String> cacheIdToNameMapping = new ConcurrentHashMap<>();
```

Comment:
Why we can't use `id` to `cache descriptor` mapping? Why do we need `id` to `name` mapping and map `name` to `cache descriptor` later?

Reply:
I was not sure if it is a good idea to duplicate the entire cache descriptor instance twice, once in the name - descriptor mapping and once here. This seemed like a memory wise cheaper alternative with minimal impact to performance.

Reply:
Don't duplicate it, use the same instance

Reply:
Fixed, thanks